app/controllers/admin/ticket_scannings_controller.rb

```diff
@@ -8,9 +8,14 @@ class TicketScanningsController < Admin::BaseController
     def create
       @ticket_scanning = TicketScanning.new(physical_ticket: @physical_ticket)
       authorize! :create, @ticket_scanning
-      @ticket_scanning.save
-      redirect_to conferences_path,
-                  notice: "Ticket with token #{@physical_ticket.token} successfully scanned."
+      if @physical_ticket.ticket.registration_ticket?
+        @physical_ticket.mark_attendance
+        @ticket_scanning.save
+        redirect_to conferences_path,
+                    notice: "Ticket with token #{@physical_ticket.token} successfully scanned."
+      else
+        redirect_to conferences_path, error: 'Ticket scanned is not a registration type ticket.'
```

Comment:
Why is this an error? I mean, you are not only going to scan registration tickets, right? It could be another kind of ticket, and scanning it could mean "consuming" it.

Reply:
True, but since we don't have any feature for consuming a ticket right now therefore I'm removing this `else` condition for now.